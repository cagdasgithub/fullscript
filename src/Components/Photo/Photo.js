import React, { useState } from "react";
import Button from "@material-ui/core/Button";
import Modal from "../UI/Modal/Modal";
import EmailForm from "../../Views/EmailForm/EmailForm";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import ButtonBase from "@material-ui/core/ButtonBase";
import { withStyles } from "@material-ui/core/styles";
import Fab from "@material-ui/core/Fab";
import Icon from "@material-ui/core/Icon";

const styles = theme => ({
  root: {
    flexGrow: 1
  },
  image: {
    width: 256,
    height: 256
  },
  fab: {
    margin: theme.spacing.unit * 1.1
  },
  img: {
    margin: "auto",
    display: "block",
    maxWidth: "100%",
    maxHeight: "100%"
  }
});

const Photo = ({
  photo: {
    id,
    urls: { regular },
    links: { download },
    description,
    user: { name, location }
  },
  classes
}) => {
  const [openModal, setOpenModal] = useState(false);
  const handleEmailClick = () => {
    setOpenModal(true);
  };
  const handleModalClose = () => {
    setOpenModal(false);
  };
  const handleEmailSend = () => {
    setOpenModal(false);
  };
  return (
    <React.Fragment>
      <Grid item>
        <ButtonBase className={classes.image} >
          <img className={classes.img}  alt="complex" src={regular} />
        </ButtonBase>
      </Grid>
      <Grid item xs={12} sm container>
        <Grid item xs container direction="column" spacing={16}>
          <Grid item xs>
            <Typography gutterBottom variant="subtitle1">
              {description}
            </Typography>
            <Typography gutterBottom>{name}</Typography>
            <Typography color="textSecondary">{location}</Typography>
          </Grid>
          <Grid item>
            <Fab aria-label="Add"  href={regular} className={classes.fab}>
              <Icon>add</Icon>
            </Fab>
            <Fab
              aria-label="Add"
              onClick={handleEmailClick}
              className={classes.fab}
            >
              <Icon>email</Icon>
            </Fab>
            <Modal open={openModal} onClose={handleModalClose}>
              <EmailForm submitForm={handleEmailSend} />
            </Modal>
          </Grid>
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

export default withStyles(styles)(Photo);
