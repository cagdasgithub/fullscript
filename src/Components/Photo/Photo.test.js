import React from "react";
import renderer from "react-test-renderer";
import Photo from "./Photo";

it("renders correctly", () => {
  const photo = {
    id: "1",
    urls: { regular: "www.google.com" },
    links: { download: "www.google.com" },
    description: "test",
    user: { name: "Cagdas", location: "Ottawa" }
  };
  const photoRendered = renderer.create(<Photo photo={photo} />).toJSON();
  expect(photoRendered).toMatchSnapshot();
});
