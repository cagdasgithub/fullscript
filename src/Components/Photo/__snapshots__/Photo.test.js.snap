// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="MuiGrid-item-6"
  >
    <button
      className="MuiButtonBase-root-102 Photo-image-2"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <img
        alt="complex"
        className="Photo-img-4"
        src="www.google.com"
      />
      <span
        className="MuiTouchRipple-root-176"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-container-5 MuiGrid-item-6 MuiGrid-grid-xs-12-45 MuiGrid-grid-sm-true-47"
  >
    <div
      className="MuiGrid-container-5 MuiGrid-item-6 MuiGrid-spacing-xs-16-28 MuiGrid-direction-xs-column-8 MuiGrid-grid-xs-true-33"
    >
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-true-33"
      >
        <h6
          className="MuiTypography-root-105 MuiTypography-subtitle1-123 MuiTypography-gutterBottom-132"
        >
          test
        </h6>
        <p
          className="MuiTypography-root-105 MuiTypography-body1-114 MuiTypography-gutterBottom-132"
        >
          Cagdas
        </p>
        <p
          className="MuiTypography-root-105 MuiTypography-body1-114 MuiTypography-colorTextSecondary-138"
        >
          Ottawa
        </p>
      </div>
      <div
        className="MuiGrid-item-6"
      >
        <a
          aria-label="Add"
          className="MuiButtonBase-root-102 MuiFab-root-141 Photo-fab-3"
          href="www.google.com"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiFab-label-142"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-151"
            >
              add
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-176"
          />
        </a>
        <button
          aria-label="Add"
          className="MuiButtonBase-root-102 MuiFab-root-141 Photo-fab-3"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiFab-label-142"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-151"
            >
              email
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-176"
          />
        </button>
      </div>
    </div>
  </div>,
]
`;
